### PR TITLE
Update API conformance

### PR DESCRIPTION
**Pull request by matthewhanson**
- Adds conformsTo links to provider catalogs
- updates search link `type` field to `application/geo+json`
- adds `method` field to provider search link with value of 'GET'
- adds additional provider search link with method=POST

### CODE REVIEW

File: search/lib/api/provider.js [hunk 1]
```diff
@@ -53,7 +55,14 @@ async function getProvider (request, response) {
       title: providerId,
       description: `Root catalog for ${providerId}`,
       stac_version: settings.stac.version,
-      links: [...links, ...childLinks]
+      links: [...links, ...childLinks],
+      conformsTo: [
+        'https://api.stacspec.org/v1.0.0-beta.1/core',
```
Comment (daniel-zamora) [inline search/lib/api/provider.js:60]: https://api.stacspec.org/v1.0.0-beta.1/core
https://api.stacspec.org/v1.0.0-beta.1/item-search
https://github.com/radiantearth/stac-api-spec
Is listed in the ticket, I'm guessing this is the fully updated list of links?

File: search/lib/util/app.js [hunk 2]
```diff
@@ -19,14 +19,18 @@ function extractTypeFromHref (href) {
...
+const createLink = (rel, href = '', title, type = 'application/json', method = null) => {
   type = extractTypeFromHref(href) || type;
 
   if (href.includes('.native')) {
     href = href.replace('.native', '.xml');
   }
 
-  return { rel: rel, href: href, title: title, type: type };
+  const link = { rel: rel, href: href, title: title, type: type };
+  if (method) {
```
Comment (daniel-zamora) [inline search/lib/util/app.js:30]: What happens here if method is null?  That isn't an error condition, just something that can sometimes happen?

**Review by daniel-zamora — APPROVED** [review]

File: search/lib/api/provider.js (hunk 1 above)
Comment (matthewhanson) [reply search/lib/api/provider.js:60]: The first 2 are the conformance classes for STAC, while the others are if you implement all the endpoints for an OGC Features API which we have here.

File: search/lib/util/app.js (hunk 2 above)
Comment (matthewhanson) [reply search/lib/util/app.js:30]: If method isn't passed into the function that it's not added to the link at all.

**Pull request merged**